src/pal/src/arch/arm/exceptionhelper.S

```diff
@@ -3,11 +3,32 @@
 // See the LICENSE file in the project root for more information.
 
 #include "unixasmmacros.inc"
+#include "asmconstants.h"
 
 .syntax unified
 .thumb
 
+// EXTERN_C void ThrowExceptionFromContextInternal(CONTEXT* context, PAL_SEHException* ex);
 LEAF_ENTRY ThrowExceptionFromContextInternal, _TEXT
-       // TODO: Implement
-       bx lr
+// Ported from src/pal/src/arch/i386/exceptionhelper.S
+	push_nonvol_reg {r7} /* FP. x64-RBP */
+
+	ldr	r4,	[r0, #(CONTEXT_R4)]
+	ldr	r5,	[r0, #(CONTEXT_R5)]
+	ldr	r6,	[r0, #(CONTEXT_R6)]
+	ldr	r7,	[r0, #(CONTEXT_R7)]
+	ldr	r8,	[r0, #(CONTEXT_R8)]
+	ldr	r9,	[r0, #(CONTEXT_R9)]
+	ldr	r10,	[r0, #(CONTEXT_R10)]
+	ldr	r11,	[r0, #(CONTEXT_R11)]
+	ldr	sp,	[r0, #(CONTEXT_Sp)]
+	ldr	lr,	[r0, #(CONTEXT_Lr)]
+
+	ldr	r2,	[r0, #(CONTEXT_Pc)]
+
+        // Store return address to the stack
+        push	{r2}
```

Comment:
@janvori Yup. Edited to be indented w/ 4 spaces. Maybe we can adopt something like Linux-Kernel's scripts/checkpatch.pl and automate style checking process later. Note that we cannot adopt it directly because it is GPLv2.